kornia/contrib/models/rt_detr/architecture/resnet_d.py

```diff
@@ -0,0 +1,87 @@
+"""ResNet-D as described in https://arxiv.org/abs/1812.01187.
+
+Based on the code from https://github.com/pytorch/vision/blob/v0.15.1/torchvision/models/resnet.py
+https://github.com/PaddlePaddle/PaddleDetection/blob/release/2.6/ppdet/modeling/backbones/resnet.py
+"""
+from __future__ import annotations
+
+from torch import nn
+
+from kornia.contrib.models.common import ConvNormAct
+from kornia.core import Module, Tensor
+from kornia.core.check import KORNIA_CHECK
+
+
+class BottleNeckD(Module):
+    expansion = 4
```

Comment:
Just follow Torchvision / PaddleDetection implementation

https://github.com/pytorch/vision/blob/main/torchvision/models/resnet.py#L115
https://github.com/PaddlePaddle/PaddleDetection/blob/release/2.6/ppdet/modeling/backbones/resnet.py#L263

Reply:
weirdo decision, but ok 😄

Reply:
I think it is because ResNet-18 and ResNet-34 use BasicBlock, which does not use expansion (expansion=1), thus this serves as a way for ResNet constructor know the output width.